projects/plugins/social/src/class-share-limits.php

```diff
@@ -36,23 +39,93 @@ public function __construct( $connections, $shares_remaining ) {
 		$this->shares_remaining = $shares_remaining;
 	}
 
+	/**
+	 * Connections are stored in a two-dimensional array, with the first dimension being the service name and the second an array of connections. We need to
+	 * flatten this array so that we can get the number of connections.
+	 *
+	 * @return int Number of connections.
+	 */
+	public function get_number_of_connections() {
+		return count( array_merge( ...array_values( $this->connections ) ) );
+	}
+
 	/**
 	 * Run functionality required to enforce sharing limits.
 	 */
 	public function enforce_share_limits() {
-		$this->maybe_disable_publicize_services_by_default();
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_classic_editor_scripts' ) );
+
+		if ( ! $this->has_more_shares_than_connections() ) {
+			/**
+			 * If the number of connections is greater than the share limit, we set all
+			 * connections to disabled by default. This allows the user to pick and
+			 * choose which services they want to share to, without going over the limit.
+			 */
+			add_filter( 'publicize_checkbox_default', '__return_false' );
+		}
+	}
+
+	/**
+	 * Check if a site has more shares than connections.
+	 *
+	 * @return bool True if there's more shares than connections left, false otherwise.
+	 */
+	public function has_more_shares_than_connections() {
+		return $this->shares_remaining >= $this->get_number_of_connections();
 	}
 
 	/**
-	 * If the number of connections is greater than the share limit, we set all
-	 * connections to disabled by default. This allows the user to pick and
-	 * choose which services they want to share to, without going over the limit.
+	 * Enqueue scripts for the Classic Editor on the post edit screen.
 	 */
-	public function maybe_disable_publicize_services_by_default() {
-		if ( $this->shares_remaining >= count( $this->connections ) ) {
+	public function enqueue_classic_editor_scripts() {
+		$current_screen = get_current_screen();
+
+		if ( $current_screen->base !== 'post' ) {
+			return;
+		}
+
+		if ( method_exists( $current_screen, 'is_block_editor' ) && $current_screen->is_block_editor() ) {
```

Comment:
Is this for older versions of WP? `is_block_editor` was added in 5.0, so it should exist.

Reply:
[86655dc](https://github.com/Automattic/jetpack/pull/26039/commits/86655dc0fb2c62081703b9b6d5791d1a7b61431d)